router/userRouter.js

```diff
@@ -11,10 +11,56 @@ import registerRules from "../validation/registerRules.js";
 
 const router = express.Router();
 
-router.get("/", getUsers);
-router.post("/add", validate(registerRules), addUser);
-router.post("/login", loginUser);
-router.put("/update", updateUser);
-router.delete("/delete", deleteUser);
+// router.get("/", getUsers);
+// router.post("/add", validate(registerRules), addUser);
+// router.post("/login", loginUser);
+// router.put("/update", updateUser);
+// router.delete("/delete", deleteUser);
+
+
+// ------ #issue5-endpoints work on 16/11/2021
+// just experimenting w/ endpoints --> this is not functional and will not run
+
+// routes to browse events without being a registered and logged in user
+router.get("/", getEventsForToday); // landing page
+router.get(`/events/${eventId}`, getEvent(eventId)); // click on one event
+router.get(`/events?category=${category}`, getEvents(category)); // filter events by 1 category
```

Comment:
Query string parameters like `/user/events?category=foo` will be accessible inside getEvents like so;

```
function getEvents(req, res) {
    console.log(req.query); // see this for values
    res.send("hi");
}
```

Because of this, you would actually have just one route, maybe `/user/events` that is handled by `getEvents`. Then inside `getEvents` you would check `req.query and check in the getEvents function if you find various values like `category=123&keyword=salsa` rather than having separate routes for each of them

Reply:
very cool, that shortens the list of endpoints considerably 👍